src/test/java/com/ericsson/ei/subscription/SubscriptionRestAPITest.java

```diff
@@ -206,74 +201,105 @@ public void updateSubscriptionFailWhenSubscriptionDoNotExist() throws Exception
-        assertEquals("[]", result.getResponse().getContentAsString());
+        JSONArray notFoundSubscriptions = new JSONObject(result.getResponse().getContentAsString()).getJSONArray("notFoundSubscriptions");
+
+        assertEquals(HttpStatus.NOT_FOUND.value(), result.getResponse().getStatus());
+        assertEquals("Subscription_Test_Not_Found", notFoundSubscriptions.get(0).toString());
     }
 
     @Test
-    public void deleteSubScriptionByName() throws Exception {
+    public void deleteSubscriptionByName() throws Exception {
         Mockito.when(subscriptionService.deleteSubscription(Mockito.anyString())).thenReturn(true);
 
-        // Send subscription as body to /subscriptions
         RequestBuilder requestBuilder = MockMvcRequestBuilders.delete("/subscriptions/Subscription_Test")
                 .accept(MediaType.APPLICATION_JSON);
 
         MvcResult result = mockMvc.perform(requestBuilder).andReturn();
 
-        SubscriptionResponse subscriptionResponse = mapper
-                .readValue(result.getResponse().getContentAsString().toString(), SubscriptionResponse.class);
         assertEquals(HttpStatus.OK.value(), result.getResponse().getStatus());
-        assertEquals("Deleted Successfully", subscriptionResponse.getMsg());
     }
 
     @Test
-    public void deleteSubScriptionByNameNotFound() throws Exception {
+    public void deleteSubscriptionByNameMulti() throws Exception {
+        Mockito.when(subscriptionService.deleteSubscription(Mockito.anyString())).thenReturn(true);
+
+        RequestBuilder requestBuilder = MockMvcRequestBuilders
+            .delete("/subscriptions/Subscription_Test,Subscription_Test_Multi,Subscription_Test_Modify")
+            .accept(MediaType.APPLICATION_JSON);
+
+        MvcResult result = mockMvc.perform(requestBuilder).andReturn();
+
+        assertEquals(HttpStatus.OK.value(), result.getResponse().getStatus());
+    }
+
+    @Test
+    public void deleteSubscriptionByNameNotFound() throws Exception {
         Mockito.when(subscriptionService.deleteSubscription(Mockito.anyString())).thenReturn(false);
 
-        // Send subscription as body to /subscriptions
         RequestBuilder requestBuilder = MockMvcRequestBuilders.delete("/subscriptions/Subscription_Test")
                 .accept(MediaType.APPLICATION_JSON);
 
         MvcResult result = mockMvc.perform(requestBuilder).andReturn();
 
-        SubscriptionResponse subscriptionResponse = mapper
-                .readValue(result.getResponse().getContentAsString().toString(), SubscriptionResponse.class);
+        JSONObject responseBody = new JSONArray(result.getResponse().getContentAsString()).getJSONObject(0);
+
         assertEquals(HttpStatus.BAD_REQUEST.value(), result.getResponse().getStatus());
-        assertEquals("Record not found for delete", subscriptionResponse.getMsg());
+        assertEquals("Subscription is not found", responseBody.getString(REASON_FIELD));
     }
```

Comment:
I think these Return messages is for user and can be improved/changed without any extra notice and this test would then fail.
Should we really be program against user dedicated text/messages(that might/probably be changed/improved.) ?

Reply:
I agree. Fixed.